### PR TITLE
[MISC] Reusable workflow

### PR DESCRIPTION
**Pull request by dragomirp**
Switch to the DPW reusable workflow

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -22,90 +22,45 @@ on:
...
-          - legacy-client-relation-integration
-          - upgrade-integration
-        agent-versions: 
-          - "2.9.45"  # renovate: latest juju 2
-          - "3.1.6"  # renovate: latest juju 3
-    name: ${{ matrix.tox-environments }} | ${{ matrix.agent-versions }}
+        juju:
+          - agent: 2.9.46
+            libjuju: ^2
+          - agent: 3.1.6
```
Comment (dragomirp) [inline .github/workflows/ci.yaml:54]: Backend relation fails on 3.1.7. Will investigate separately.

File: tests/integration/relations/test_db_admin.py [hunk 2]
```diff
@@ -22,14 +23,14 @@
 
 logger = logging.getLogger(__name__)
 
-FIRST_DISCOURSE_APP_NAME = "discourse-k8s"
```
Comment (dragomirp) [inline tests/integration/relations/test_db_admin.py:25]: discourse-k8s was ported to the new interface.

Comment (taurus-forever) [reply tests/integration/relations/test_db_admin.py:25]: Crazy naming choise... BTW, is "discourse-k8s" R.I.P? Why do we need two identical charms? o_0

**Review by taurus-forever — APPROVED** [review]
Well done. THANK YOU!

File: .github/renovate.json5 [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "extends": ["github>canonical/data-platform//renovate_presets/charm.json5"],
+  "reviewers": ["dragomirp", "marceloneppel", "taurus-forever"],
+  "packageRules": [
+    // Later rules override earlier rules
+    {
+      "matchPackageNames": ["pydantic"],
+      "allowedVersions": "<2.0.0"
+    }, {
```
[the remark this answers is not in the document]
Comment (dragomirp) [reply .github/renovate.json5:10]: I don't consider this a temporary issue. There are multiple charm libs that will need to be updated to work with pydantic 2 and it also fails to compile with focal's rustc. I doubt we can make the jump anytime soon.

File: .github/renovate.json5 [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+      "matchPackageNames": ["python"],
+      "allowedVersions": "<3.11"
```
[the remark this answers is not in the document]
Comment (dragomirp) [reply .github/renovate.json5:12]: The python version is charm specific, it will vary depending on the base.

**Review by marceloneppel — APPROVED** [review]

File: .github/workflows/lib-check.yaml [hunk 5]
```diff
@@ -26,6 +26,16 @@ jobs:
...
+      - name: Set up environment
+        run: |
+          pipx install tox
+          pipx install poetry
+          pipx inject poetry poetry-plugin-export
+          # TODO: Remove after https://github.com/python-poetry/poetry/pull/5980 is closed
+          poetry config warnings.export false
+      - name: Generate requirements
+        run: |
+          tox run -e generate-requirements
```
Comment (dragomirp) [inline .github/workflows/lib-check.yaml:38]: Charmcraft complains if there are no requirements.txt when fetching libs. Maybe we should move this to the dpw?

**Comment by dragomirp** [comment]
Enabled `charm-strict-dependencies`. Please, take another look.

File: .github/workflows/lib-check.yaml (hunk 5 above)
Comment (dragomirp) [reply .github/workflows/lib-check.yaml:38]: Touch doesn't seem to work locally for me. Possibly because PGB has `charm-binary-python-packages`.

Comment (dragomirp) [reply .github/workflows/lib-check.yaml:38]: That will rebuild the charm again in the lib-check workflow.

Comment (dragomirp) [reply .github/workflows/lib-check.yaml]: Tried it, but it moves the requirements.txt on exit: https://github.com/canonical/pgbouncer-k8s-operator/actions/runs/7449186139/job/20265256079?pr=168

File: pyproject.toml [hunk 6]
```diff
@@ -22,7 +22,9 @@ poetry-core = "^1.8.1"
 lightkube = "^0.15.0"
 lightkube-models = "^1.29.0.6"
 pydantic = "^1.10.13"
-# psycopg2 = "^2.9.5" # Injected in charmcraft.yaml
+psycopg2-binary = "^2.9.9"
+psycopg-binary = "^3.1.17"
+psycopg = {extras = ["binary"], version = "^3.1.17"}
```
[the remark this answers is not in the document]
Comment (dragomirp) [reply pyproject.toml]: Looks like no.

**Review by taurus-forever — APPROVED** [review]

**Pull request merged**